src/SB/Core/gc/iModel.cpp

```diff
@@ -26,51 +47,314 @@ void* GetHierarchy(RpAtomic* frame)
+
+    if (model == 0)
+    {
+        return 0;
+    }
+    else
+    {
+		clump = model->clump;
+        nextModel = model;
+        RpClumpForAllAtomics(clump, NextAtomicCallback, &nextModel);
+		return nextModel;
+    }
+}
+
+U32 iModelNumBones(RpAtomic* model)
+{
+    RpHAnimHierarchy* obj = (RpHAnimHierarchy*)GetHierarchy(model);
+    return obj != 0 ? (U32)obj->numNodes : 0;
+}
+
+void iModelQuatToMat(xQuat* q, xVec3* a, RwMatrixTag* t)
 {
+    q->s = -q->s;
+    xQuatToMat(q, (xMat3x3*)t);
+    q->s = -q->s;
+    t->pos.x = a->x;
+    t->pos.y = a->y;
+    t->pos.z = a->z;
 }
+
+// WIP
+void iModelAnimMatrices(RpAtomic* model, xQuat* quat, xVec3* tran, RwMatrixTag* mat)
+{
+    RwMatrixTag* pMatrixArray;
+    RpHAnimNodeInfo* iVar1;
+    RwMatrixTag matrixStack [33];
+    U32 pCurrentFrameFlags;
+    RpHAnimNodeInfo* pCurrentFrame;
+    int numFrames;
+    RwMatrixTag* pMatrixStackTop;
+
+    pCurrentFrame = (RpHAnimNodeInfo *)GetHierarchy(model);
+
+    if (pCurrentFrame != NULL)
+    {
+	    pMatrixStackTop = &matrixStack[0];
+        pMatrixStackTop->at.z = 1.0;
+
+        matrixStack[0].up.y = 1.0;
+        matrixStack[0].right.x = 1.0;
+        matrixStack[0].up.x = 0.0;
+        matrixStack[0].right.z = 0.0;
+        matrixStack[0].right.y = 0.0;
+        matrixStack[0].at.y = 0.0;
+        matrixStack[0].at.x = 0.0;
+        matrixStack[0].up.z = 0.0;
+        matrixStack[0].pos.z = 0.0;
+        matrixStack[0].pos.y = 0.0;
+        matrixStack[0].pos.x = 0.0;
+        matrixStack[0].flags |= 0x20003;
+
+		matrixStack[1] = matrixStack[0];
+        numFrames = pCurrentFrame->nodeIndex;
+        pMatrixArray = (&matrixStack[1]);
+        iVar1 = (RpHAnimNodeInfo *)pCurrentFrame[1].nodeID;
+
+		pMatrixArray++;
+		for (int i = 0; i < numFrames; i++)
+		{
+			pCurrentFrameFlags = iVar1->flags;
+            if ((pCurrentFrameFlags & 2) != 0)
+            {
+				*pMatrixArray++ = matrixStack[0];
+            }
+
+			RwMatrixTag afStack_8e8;
+            iModelQuatToMat(quat, tran, &afStack_8e8);
+
+			RwMatrixTag auStack_8a8;
+            xMat4x3Mul((xMat4x3*)&auStack_8a8,(xMat4x3*)&afStack_8e8, (xMat4x3*)&matrixStack[0]);
+
+			*mat = auStack_8a8;
+            if ((pCurrentFrameFlags & 1) != 0)
+            {
+				pMatrixArray = &pMatrixArray[-1];
+                pMatrixStackTop = pMatrixArray;
+            }
+            else
+            {
+                pMatrixStackTop = (RwMatrixTag *)&auStack_8a8;
+            }
+            matrixStack[0] = *pMatrixStackTop;
+            mat++;
+            quat++;
+            tran++;
+            iVar1++;
+		}
+    }
+}
+
+RpAtomic* iModelCacheAtomic(RpAtomic* model)
+{
+	return model;
+}
+
+void iModelRender(RpAtomic* model, RwMatrixTag* mat)
+{
+    RpHAnimHierarchy *iVar1;
+    RpGeometry* geom;
+    RwMatrixTag *pAnimOldMatrix;
+    RwFrame* frame;
+
+    iVar1 = (RpHAnimHierarchy *)GetHierarchy(model);
+    if ((S8)init == false)
+    {
+        draw_all = 1;
+        init = true;
+    }
```

Comment:
```suggestion
    static int draw_all = 1;
```

Reply:
Yep, `init` is an implicit variable generated by the compiler to handle the static initialization